src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -551,7 +551,12 @@ BlobMetadata buildAndUpload(String blobPath, List<List<ChannelData<T>>> blobData
     Timer.Context buildContext = Utils.createTimerContext(this.owningClient.buildLatency);
 
     // Construct the blob along with the metadata of the blob
-    BlobBuilder.Blob blob = BlobBuilder.constructBlobAndMetadata(blobPath, blobData, bdecVersion);
+    BlobBuilder.Blob blob =
+        BlobBuilder.constructBlobAndMetadata(
+            blobPath,
+            blobData,
+            bdecVersion,
+            this.owningClient.getParameterProvider().getEnableChunkEncryption());
```

Comment:
we should just check if the client is for iceberg or not?

Reply:
I asked alec to not pass around isIceberg everywhere - we want one place (in the parameter provider) that looks at isIceberg and sets up all the configuration / feature flags in that one place.
This is because over time we'll have a proliferation of `if (isIceberg)` in ten different places, making it incredibly hard to figure out what all is different for iceberg vs. fdn code paths.

At the same time I don't mean to say all these features need to be controllable by customers, I remember leavinga comment in the last round that enableChunkEncryption should not be public to customers + should not be publically settable/overrideable.

Reply:
Made non-settable to customers. For `getEnableChunkEncrption()` I think it should still be public as the `ParameterProvider` is located in different package with `FlushService`

Reply:
i see, but `ParameterProvider` is supposed to contain only configurable parameters by customers, do you want to create a separate class for internal control? (and this class can be in the same package as FlushService)

Reply:
Interesting, maybe we should chat f2f on this, but i don't see where ParameterProvider object is returned out to customers?
I see IngestClientInternal has a non-public getParameterProvider method, and the SnowflakeStreamingIngestClient interface also does not have any methods to expose the parameter provider.

If/when we do want to expose this object out in future, we can easily introduce an interface IParameterProvider with only the methods that we want to expose out, and add a getter on client that returns objects of this interface?

Reply:
We're not exposing ParameterProvider directly, but if you see how the ParameterProvider is created, it will take an user input property/parameter overrides as part of the client creation.